### PR TITLE
Add Databricks dialect support for Unity Catalog

### LINKED ISSUE #4567: Add SparkSQL dialect support for Databricks Unity Catalog
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

We are using the [Unity Catalog](https://docs.databricks.com/data-governance/unity-catalog/index.html) feature of Databricks at our company. This introduces the concept of [Catalogs](https://docs.databricks.com/data-governance/unity-catalog/create-catalogs.html) as object types.

Catalogs are high-level data stores that contain Hive-style Databases (aka Schemas), and can be created and managed using common DDL grammar.

### Use case

We want to be able to use SQLFluff to format and lint our Databricks SQL notebooks, which are using common Unity Catalog DDL statements. I've provided some example statements that we might execute when running automated test code for testing ETL logic:

`CREATE CATALOG test;`
`ALTER CATALOG test SET OWNER TO '3b3bb68a-6249-47ec-b80b-8d6f5f97f0b0';`
`USE CATALOG test;`
`CREATE SCHEMA run_83264583;`
`USE SCHEMA run_83264583;`
`<run some test cases>`
`DROP SCHEMA run_83264583 CASCADE;`
`DROP CATALOG test;`

### Dialect

databricks (sparksql)

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by liamperritt**
### Brief summary of the change made

`fixes #4567`

Add ~~SparkSQL~~ Databricks dialect support for [Unity Catalog](https://docs.databricks.com/data-governance/unity-catalog/index.html). Specifically, add support for DDL statements that create and manage Unity Catalog [catalogs](https://docs.databricks.com/data-governance/unity-catalog/create-catalogs.html).

### Are there any other side effects of this change that we should be aware of?
I'm not aware of any potential side effects. Changes are limited to the SparkSQL and Databricks dialects.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_sparksql.py [hunk 1]
```diff
@@ -805,8 +805,38 @@ class DatatypeSegment(PrimitiveTypeSegment):
     )
 
 
+# Object Reference Segments
+class CatalogReferenceSegment(ansi.ObjectReferenceSegment):
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_sparksql.py:809]: can you add this content to dialect_databricks, and not spark?

File: src/sqlfluff/dialects/dialect_databricks.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
 sparksql_dialect = load_raw_dialect("sparksql")
 databricks_dialect = sparksql_dialect.copy_as("databricks")
 
+databricks_dialect.sets("unreserved_keywords").update(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_databricks.py:15]: this isn't needed as this set is already the spark unreserved keywords, due to the copy_as above

File: src/sqlfluff/dialects/dialect_sparksql.py (hunk 1 above)
Comment (liamperritt) [reply src/sqlfluff/dialects/dialect_sparksql.py]: Thanks for the direction! I've moved all 'catalog'-related content to the Databricks dialect.

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**